src/app/portlets/content-types/edit/content-types-edit.component.spec.ts

```diff
@@ -39,85 +46,167 @@ class TestContentTypeLayoutComponent {
-                            path: 'test'
-                        }
-                    ])
-                ],
-                providers: [
-                    { provide: LoginService, useClass: LoginServiceMock },
-                    {
-                        provide: ActivatedRoute,
-                        useValue: { params: Observable.from([{ id: '1234' }]) }
-                    },
-                    CrudService,
-                    FieldService,
-                    Location
-                ]
-            });
+            DOTTestBed.configureTestingModule(configCreateMode);
 
             fixture = DOTTestBed.createComponent(ContentTypesEditComponent);
             comp = fixture.componentInstance;
             de = fixture.debugElement;
             el = de.nativeElement;
-            route = fixture.debugElement.injector.get(ActivatedRoute);
 
             crudService = fixture.debugElement.injector.get(CrudService);
             location = fixture.debugElement.injector.get(Location);
+            dotRouterService = fixture.debugElement.injector.get(DotRouterService);
+
+            fixture.detectChanges();
         })
     );
 
-    it('should have Content Types Layout', () => {
+    it('should has dialog opened by default', () => {
+        const dialog = de.query(By.css('p-dialog'));
+        expect(dialog).not.toBeNull();
+        expect(dialog.componentInstance.visible).toBeTruthy();
+    });
+
+    it('should have cancel button', () => {
+        const cancelButton = de.query(By.css('.content-type__cancel'));
+        expect(cancelButton === null).toBe(false);
+    });
+
+    it('should have save button and disabled by default', () => {
+        const saveButton = de.query(By.css('.content-type__save'));
+        expect(saveButton === null).toBe(false);
+        expect(saveButton.nativeElement.disabled).toBe(true);
+    });
+
+    it('should submit form when save button is clicked', () => {
+        const form: ContentTypesFormComponent = de.query(By.css('dot-content-types-form')).componentInstance;
+        spyOn(form, 'submitForm');
+        form.canSave = true;
+        fixture.detectChanges();
+
+        const saveButton = de.query(By.css('.content-type__save'));
+        saveButton.nativeElement.click();
+        expect(form.submitForm).toHaveBeenCalledTimes(1);
+    });
+
+    it('should bind save button disabled attribute to canSave property from the form', () => {
+        const form: ContentTypesFormComponent = de.query(By.css('dot-content-types-form')).componentInstance;
+        form.canSave = true;
+        fixture.detectChanges();
+
+        const saveButton = de.query(By.css('.content-type__save'));
+        expect(saveButton === null).toBe(false);
+        expect(saveButton.nativeElement.disabled).toBe(false);
+    });
+
+    it('should close the dialog and redirect', () => {
+        spyOn(dotRouterService, 'gotoPortlet');
+        spyOn(comp, 'cancelForm').and.callThrough();
```

Comment:
we dont should test that this method is called, we should test that the dialog is closed

Reply:
I do validate that the diablog is close by doing this:

```expect(comp.show).toBe(false);```

Also there is other test where I validate that the dialog show or hide state is binded to the show property, like in this 3

https://github.com/dotCMS/core-web/blob/issue-13330-content-types-form-restyling/src/app/portlets/content-types/edit/content-types-edit.component.spec.ts#L129

https://github.com/dotCMS/core-web/blob/issue-13330-content-types-form-restyling/src/app/portlets/content-types/edit/content-types-edit.component.spec.ts#L185

https://github.com/dotCMS/core-web/blob/issue-13330-content-types-form-restyling/src/app/portlets/content-types/edit/content-types-edit.component.spec.ts#L192